overlord/state/taskrunner.go

```diff
@@ -193,10 +208,7 @@ func (r *TaskRunner) Stop() {
 		tb.Kill(nil)
 	}
 
-	for id, tb := range r.tombs {
-		tb.Wait()
-		delete(r.tombs, id)
-	}
+	r.wait()
```

Comment:
I'm slightly concerned about that Ensure running in the background in those tests. That goroutine can freeze until the whole test is done, and trigger new goroutines to fire. We should probably have a stopped flag which says "Alright, at this point we have no background activity and we're not taking any further work on this runner!", so we bail when Ensure sees the flag.